src/var.cpp

```diff
@@ -1436,6 +1436,10 @@ void jitc_var_read(uint32_t index, size_t offset, void *dst) {
     if (v->is_literal() || v->is_undefined()) {
         memcpy(dst, &v->literal, isize);
     } else if (v->is_evaluated()) {
+        if (jitc_flags() & (uint32_t) JitFlag::FreezingScope)
+            jitc_raise("jit_var_read(): reading from evaluated variables while "
```

Comment:
@wjakob Under the new policy of never crashing the kernel and including more troubleshooting info, how could we make it easy for the user to find the source of this illegal `jit_var_read()`?
Would something like "enable the Debug flag and we will show the Python line that triggers this" be possible?

Reply:
The exception will bubble up through the C++ code and then be re-raised from Python. As far as I can see, this is already good enough.